src/core/include/openvino/core/except.hpp

```diff
@@ -145,7 +145,7 @@ class OPENVINO_API NotImplemented : public AssertFailure {
 //
 #define OPENVINO_ASSERT_HELPER2(exc_class, ctx, check, ...)                      \
     do {                                                                         \
-        if (!(check)) {                                                          \
+        if (!static_cast<bool>(check)) {                                         \
```

Comment:
What's wrong with `!(expr)`? Has it become ambiguous since C++17?

Reply:
But it's redundant - `!` operator implies casting operand to `bool`. Moreover the complex `check` expressions clang-tidy complains about evaluate to `bool` as well.

Reply:
The main issue here is a macro.
The macro does not allow us to fully control the structure of the underlying code. We could silence the warning with NOLINT here, but we can get around it by making bool cast a bit more explicit, we should not be an issue, considering the fact that this is localized within like 4th level of the macro expansion which people basically never look at.
I am personally fine with both NOLINT and static_cast